src/TimeModeling/LinearOperators/lazy.jl

```diff
@@ -178,17 +178,32 @@ end
 
 ############################################################################################################################
 ###### Inpout processing for propagation
-make_src(q, P::jAdjoint{judiProjection{D}}) where D = (P.data[1], make_input(q))
-make_src(q, P::jAdjoint{judiLRWF{D}}) where D = (make_input(q), P.data[1])
-make_src(q, P::judiProjection{D}) where D = (P.data[1], make_input(q))
+function make_src(q::judiMultiSourceVector, P::judiProjection{D}) where D
+    P.data[1] ≈ q.geometry[1] || throw(ArgumentError("Incompatible geometry and source"))
+    return P.data[1], make_input(q)
+end
+
+make_src(q, P::judiProjection{D}) where D = (P.data[1], reshape(make_input(q), P.data.nt[1], P.data.nrec[1]))
 make_src(q, P::judiLRWF{D}) where D = (make_input(q), P.data[1])
-make_src(rhs::judiRHS) = make_src(rhs.d, rhs.P)
+make_src(rhs::judiWavelet) = make_src(rhs.d, rhs.P)
+make_src(q, P::jAdjoint{<:judiNoopOperator}) = make_src(q, P.op)
 
 get_nsrc(P::Projection) = P.m[:src]
 get_nsrc(P::jAdjoint{<:Projection}) = P.op.m[:src]
 get_nt(P::Projection) = P.n[:time]
 get_nt(P::jAdjoint{<:Projection}) = P.op.n[:time]
 
+function reshape(x::Vector{T}, P::judiProjection{T}, ::Model; with_batch=false) where T
+    out = reshape(x, P.geometry)
+    out = with_batch ? reshape(out, size(out, 1), size(out, 2), 1, size(out, 3)) : out
+    return out
+end
+
+function reshape(x::Vector{T}, P::judiWavelet{T}, m::Model; with_batch=false) where T
+    out = with_batch ? reshape(x, m.n..., 1,get_nsrc(P)) : reshape(x, m.n..., get_nsrc(P))
```

Comment:
same here if we just need `get_nsrc(P)` why don't we `reshape(::Vector, ::Integer, m::Model)`

Reply:
`get_nsrc(P)` is defined for any projection, wavelet or geometry, so you wouldn't know where you're supposed to be.